File: pattern8/sort_in_java_v2/Character.java
package pattern8.sort_in_java_v2;

class Character implements Comparable {
    String name;
    int attack;

    public Character(String name, int attack) {
        this.name = name;
        this.attack = attack;
    }

    @Override
    public int compareTo(Object o) {
        Character other = (Character) o;
        if (this.attack > other.attack) {
            return 1;
        } else if (this.attack == other.attack) {
            return 0;
        } else {
            return -1;
        }
    }

    @Override
    public String toString() {
        return "Character(" + this.name + "," + this.attack + ")";
    }

}

File: pattern8/sort_in_python/character.py
class Character:
    def __init__(self, name: str, attack: int) -> None:
        self.name = name
        self.attack = attack

    def __str__(self) -> str:
        return "Character({},{})".format(self.name, self.attack)

    def __repr__(self) -> str:
        return self.__str__()
